### PR TITLE
Add group API locks

### PR DESCRIPTION
**Pull request by gabrieljackson**
This change does the following:
 - Group update and delete API endpoints now lock the group object
   before making changes.
 - Removed group version API endpoint. The update endpoint handles
   version and other group configuration changes.
 - Refactored group request objects and streamlined validation.

https://mattermost.atlassian.net/browse/MM-23442

### CODE REVIEW

File: internal/api/group.go [hunk 1]
```diff
@@ -21,54 +21,28 @@ func initGroup(apiRouter *mux.Router, context *Context) {
...
 	groupRouter.Handle("", addContext(handleUpdateGroup)).Methods("PUT")
 	groupRouter.Handle("", addContext(handleDeleteGroup)).Methods("DELETE")
-
-	groupRouter.Handle("/mattermost/{version:[A-Za-z0-9.]+}", addContext(handleChangeGroupMattermostVersion)).Methods("PUT")
 }
 
-// handleChangeGroupMattermostVersion responds to PUT
-// /api/group/{group}/mattermost/{version}, upgrading or downgrading
-// the installation to the Mattermost version embedded in the request.
-func handleChangeGroupMattermostVersion(c *Context, w http.ResponseWriter, r *http.Request) {
```
Comment (gigawhitlocks) [inline internal/api/group.go]: I guess the right way to change this setting is now through using `handleUpdateGroup`?

**Comment by jwilander** [comment]
I'll let @gigawhitlocks handle the dev review

File: model/group_request.go [hunk 2]
```diff
@@ -25,18 +25,22 @@ func NewCreateGroupRequestFromReader(reader io.Reader) (*CreateGroupRequest, err
 		return nil, errors.Wrap(err, "failed to decode create group request")
 	}
 
-	if createGroupRequest.Name == "" {
-		return nil, errors.New("must specify name")
-	}
-	if createGroupRequest.Version == "" {
-		return nil, errors.New("must specify version")
+	err = createGroupRequest.Validate()
```
Comment (gigawhitlocks) [inline model/group_request.go]: :+1: I feel like you told me to move this line here during my PR so my apologies for missing it, assuming that's what happened. Regardless, I'm happy to see this moved.

File: internal/api/group.go (hunk 1 above)
Comment (gabrieljackson) [reply internal/api/group.go]: Yep, that's exactly right! The version endpoint was from the original group code where we didn't see a need for anything other than version so we streamlined it.

The cloud client hasn't actually used that endpoint in a while.

File: model/group_request.go (hunk 2 above)
Comment (gabrieljackson) [reply model/group_request.go]: No worries! It's there now.

**Comment by gabrieljackson** [comment]
Thanks for the review!

File: internal/api/group.go (hunk 1 above)
Comment (gigawhitlocks) [reply internal/api/group.go]: Perfect :+1:

**Pull request merged**